pkg/rule/repo.go

```diff
@@ -171,20 +174,36 @@ func (r Rule) Run(input Parameters, ctx Context, decisions []int) (Parameters, [
 		}
 	}
 
-	params := Parameters{}
+	var (
+		params = Parameters{}
+		d      = decisions
+	)
+
+	var diceRollout int
+	if len(decisions) > 0 {
+		diceRollout = decisions[0]
+	} else {
+		diceRollout = randInt()
```

Comment:
We could remove this branch by declaring `diceRollout` with the random number and override if a decision is present.